df.items.xml

```diff
@@ -623,46 +631,60 @@
                 <pointer name='u' type-name='unit'/>
                 <pointer name='j' type-name='job'/>
             </vmethod>
-            <vmethod><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/></vmethod>
-            <vmethod><int32_t/></vmethod>
+            <vmethod name='addImprovements'>
```

Comment:
`addImprovement`? I only see one

Reply:
the code may be able to add more than one improvement, and it is parallel with the subsequent method which definitely can add multiple improvements

Reply:
ok, i reviewed the one non-trivial implementation and it only adds one improvement, so we'll drop the 's'